src/Plugins/ComplexCore/src/ComplexCore/Filters/FindVertexToTriangleDistancesFilter.cpp

```diff
@@ -0,0 +1,146 @@
+#include "complex/DataStructure/Geometry/VertexGeom.hpp"
+#include "complex/Filter/Actions/CreateArrayAction.hpp"
+#include "complex/Filter/Actions/DeleteDataAction.hpp"
+#include "complex/Parameters/ArrayCreationParameter.hpp"
+#include "complex/Parameters/ArraySelectionParameter.hpp"
+#include "complex/Parameters/GeometrySelectionParameter.hpp"
+
+using namespace complex;
+
+namespace
+{
+inline constexpr StringLiteral k_TriangleBounds("Triangle Bounds");
+}
+
+namespace complex
+{
+//------------------------------------------------------------------------------
+std::string FindVertexToTriangleDistancesFilter::name() const
+{
+  return FilterTraits<FindVertexToTriangleDistancesFilter>::name.str();
+}
+
+//------------------------------------------------------------------------------
+std::string FindVertexToTriangleDistancesFilter::className() const
+{
+  return FilterTraits<FindVertexToTriangleDistancesFilter>::className;
+}
+
+//------------------------------------------------------------------------------
+Uuid FindVertexToTriangleDistancesFilter::uuid() const
+{
+  return FilterTraits<FindVertexToTriangleDistancesFilter>::uuid;
+}
+
+//------------------------------------------------------------------------------
+std::string FindVertexToTriangleDistancesFilter::humanName() const
+{
+  return "Find Vertex to Triangle Distances";
+}
+
+//------------------------------------------------------------------------------
+std::vector<std::string> FindVertexToTriangleDistancesFilter::defaultTags() const
+{
+  return {"#Sampling", "#Spatial"};
+}
+
+//------------------------------------------------------------------------------
+Parameters FindVertexToTriangleDistancesFilter::parameters() const
+{
+  Parameters params;
+
+  // Create the parameter descriptors that are needed for this filter
+  params.insertSeparator(Parameters::Separator{"Required Input Arrays"});
+  params.insert(std::make_unique<GeometrySelectionParameter>(k_VertexDataContainer_Key, "Source Vertex Geometry", "The Vertex Geometry point cloud to map to triangles", DataPath{},
+                                                             GeometrySelectionParameter::AllowedTypes{IGeometry::Type::Vertex}));
+  params.insert(std::make_unique<GeometrySelectionParameter>(k_TriangleDataContainer_Key, "Target Triangle Geometry", "The triangle geometry to compare against", DataPath{},
+                                                             GeometrySelectionParameter::AllowedTypes{IGeometry::Type::Triangle}));
+  params.insert(
+      std::make_unique<ArraySelectionParameter>(k_TriangleNormalsArrayPath_Key, "Triangle Normals", "The triangle geometry's normals array", DataPath{}, std::set<DataType>{DataType::float64}));
+
+  params.insertSeparator(Parameters::Separator{"Created Output Arrays"});
+  params.insert(std::make_unique<ArrayCreationParameter>(k_DistancesArrayPath_Key, "Distances Array", "The array to store distance between vertex and triangle", DataPath{}));
```

Comment:
These might be better served as DataObjectNameParameter and then in the Preflight you can create the output arrays in the proper vertex attribute matrix that you get from the VertexGeometry.